### PR TITLE
Fix defaulting behavior of cluster factories and classes

### PR DESCRIPTION
**Pull request by dongreenberg**

### CODE REVIEW

**Comment by dongreenberg** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #377** <a href="https://app.graphite.dev/github/pr/run-house/runhouse/377?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/run-house/runhouse/377?utm_source=stack-comment).

**Review by carolineechen — COMMENTED** [review]
general comment about the way we're doing cluster factory functions right now, but could we get rid of kwargs for specific factories like ondemand/kubernetes/sagemaker? I get the point for the generic cluster class if we're passing in those args into more specific cluster factories, but for the others I think we should just include the supported variables as documented fn args and not support random additional kwargs that won't be passed in. this also made us have to be extra careful with parsing kwargs and setting constraints as we've done with the k8s factory

cc @RohanSreerama5

File: runhouse/resources/hardware/cluster_factory.py [hunk 1]
```diff
@@ -85,51 +79,27 @@ def cluster(
...
+            server_connection_type=server_connection_type,
+            ssl_keyfile=ssl_keyfile,
+            ssl_certfile=ssl_certfile,
+            den_auth=den_auth,
+            kwargs=kwargs,
         )
+        # Filter out None/default values
+        alt_options = {k: v for k, v in alt_options.items() if v is not None}
+        try:
+            c = Cluster.from_name(name, dryrun, alt_options=alt_options)
```
Comment (rohinb2) [inline runhouse/resources/hardware/cluster_factory.py:97]: we want to enable this `alt_options` behavior right? this is a changefrom before

**Comment by rohinb2** [comment]
this seems very straightforward

**Review by rohinb2 — APPROVED** [review]

File: runhouse/resources/hardware/cluster_factory.py (hunk 1 above)
Comment (dongreenberg) [reply runhouse/resources/hardware/cluster_factory.py:97]: Y

**Comment by dongreenberg** [comment]
This failure is just because I was running the tests locally at the same time so the !rh.exists failed. Merging.

**Pull request merged**